src/main.c

```diff
@@ -1,50 +1,145 @@
 #include <clang-c/Index.h>
 #include <stdio.h>
 
-void printCXString(CXString s)
+enum CXChildVisitResult visitor(CXCursor, CXCursor, CXClientData); 
+enum CXChildVisitResult enum_child_visitor(CXCursor, CXCursor, CXClientData); 
+enum CXChildVisitResult struct_child_visitor(CXCursor, CXCursor, CXClientData); 
+void enum_handler(CXCursor);
+void struct_handler(CXCursor);
+
+
+void print_CXString(CXString s)
 {
   printf("%s", clang_getCString(s));
   clang_disposeString(s);
 }
 
-void print_cursor_type(CXCursor cursor){
+void print_cursor_type(CXCursor cursor)
+{
   CXType cursor_type = clang_getCursorType(cursor);
-  printCXString(clang_getTypeSpelling(cursor_type));
+  print_CXString(clang_getTypeSpelling(cursor_type));
 }
 
-void print_cursor_spelling(CXCursor cursor){
+void print_cursor_spelling(CXCursor cursor)
+{
   CXString cursor_spelling = clang_getCursorSpelling(cursor);
-  printCXString(cursor_spelling);
+  print_CXString(cursor_spelling);
 }
 
-enum CXChildVisitResult enum_visitor(CXCursor cursor, CXCursor parent, CXClientData client_data) {
-  if(clang_getCursorKind(cursor) == CXCursor_IntegerLiteral){
-    long long enum_const_value = clang_getEnumConstantDeclValue(parent);
-    printf("IntegerLiteral\t %lld \t\tParent = ", enum_const_value);
-    print_cursor_spelling(parent);
-    printf("\n");
-  }
+// void print_cursor_kind(CXCursor cursor){
```

Comment:
Usually I wipe the code that I don't need instead of commenting it.
You have the backup in the git history if you need it.

Reply:
Ok @gligneul